### PR TITLE
Add argocd manifest to install the postgresql chart

### PR DESCRIPTION
**Pull request by cmoulliard**
Add argocd manifest to install the postgresql chart when user selects postgresql db and deploy the hlme under the app namespace

### CODE REVIEW

**Comment by cmoulliard** [comment]
Test with commit 1711f157d30ec0a3cf75c5912e4542a0877dbc5d is working as we got such an argocd file created:
```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: my-quarkus-app-db
  namespace: openshift-gitops
  finalizers:
    - resources-finalizer.argocd.argoproj.io
spec:
  project: default
  destination:
    server: https://kubernetes.default.svc
    namespace: qshift
  source:
    chart: bitnami/postgresql
    repoURL: https://charts.bitnami.com/bitnami
    targetRevision: 11.9.13
    helm:
      values: |
        auth.username: 
        auth.password: 
        auth.database: quarkus-jdbc-postgresql
  syncPolicy:
    automated:
      prune: true
      selfHeal: false
    syncOptions:
      - CreateNamespace=true
      - RespectIgnoreDifferences=true
      - ApplyOutOfSyncOnly=true
```

TODOs:
- We should fix the issue about missing parameters non passed: username and password
- helm template is not found buyt argocd
```
Failed to load target state: failed to generate manifest for source 1 of 1: rpc error: code = Unknown desc = `helm pull --destination /tmp/e284d00e-5078-466d-9d85-09c9b37a9319 --version 11.9.13 --repo https://charts.bitnami.com/bitnami bitnami/postgresql` failed exit status 1: Error: chart "bitnami/postgresql" version "11.9.13" not found in https://charts.bitnami.com/bitnami repository
```

File: qshift/templates/quarkus-application/manifests/argocd/${{values.component_id}}-db.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    server: https://kubernetes.default.svc
+    namespace: ${{ values.appNamespace }}
+  source:
+    chart: postgresql
+    repoURL: https://charts.bitnami.com/bitnami
+    targetRevision: 11.9.13
+    helm:
+      values: |
+        auth.username: ${{ values.username }}
+        auth.password: ${{ values.password }}
```
Comment (iocanel) [inline qshift/templates/quarkus-application/manifests/argocd/${{values.component_id}}-db.yaml:21]: I think that we should remove password from here. We shouldn't  be providing the password ourselves.
We should let helm use a generated one and pick ints value from the secret.

Comment (cmoulliard) [reply qshift/templates/quarkus-application/manifests/argocd/${{values.component_id}}-db.yaml]: > I think that we should remove password from here. We shouldn't be providing the password ourselves.
> We should let helm use a generated one and pick its value from the secret.

This is something that we will do in a next PR. Current bitnami helm chart dont generate a secret at all ! Primaza will help us here ;-)

**Comment by cmoulliard** [comment]
Another issue reported during installation of the Deployment resource
```
one or more objects failed to apply, reason: Deployment.apps "my-quarkus-app-deploy-quarkus-deploy" is invalid: metadata.annotations: Invalid value: "app.openshift.io/vcs-uri:https://github.com/ch007m/my-quarkus-app.gitapp.quarkus.io/quarkus-version": a qualified name must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName', or 'my.name', or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]') with an optional DNS subdomain prefix and '/' (e.g. 'example.com/MyName')
```

**Comment by cmoulliard** [comment]
This PR works except the authentication but I suspect an issue if the PVC is not deleted and recreated

**Comment by cmoulliard** [comment]
Template + helm chart + application.properties is using now the Helm postgresql generated password => secret
I did a new end to end test and it works
Can you review please ? @iocanel

**Review by iocanel — COMMENTED** [review]
Adding credentials into files that are checked into source control is not an acceptable approach. The charts do generate a secret with credentials inside it. We should pass the generated secret using something like:

```yaml
        - env:
            - name: QUARKUS_DATASOURCE_PASSWORD
              valueFrom:
                secretKeyRef:
                  key: postgres-password
                  name: postgresql-1707212401
``` 

This is something that is verified to work with `bitnami/postgresql`

**Comment by cmoulliard** [comment]
> Adding credentials into files that are checked into source control is not an acceptable approach. The charts do generate a secret with credentials inside it. We should pass the generated secret using something like:

This is also what I'm doing. The secret is generated by the helm chart, stored in a secret and secret mounted within the deployment => https://github.com/q-shift/qshift-templates/blob/32e85126d8fbe03e3e85f9d0fc5a04b8a69fc620/qshift/templates/quarkus-application/manifests/helm/deploy/templates/deployment.yaml#L38-L42

@iocanel

File: qshift/templates/quarkus-application/skeletons/quarkus-orm-rest/src/main/resources/application.properties [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
-%prod.quarkus.datasource.username=postgres
+# Kubernetes Service name to access the DB should be as defined within the helm chart => postgresql:5432
+# apiVersion: v1
+# kind: Service
+# metadata:
+#  name: {{ include "postgresql.primary.fullname" . }}
+#  see: https://github.com/bitnami/charts/blob/9ba6ba51cf7c094e457396c423f5843df46023a2/bitnami/postgresql/templates/primary/svc.yaml#L4
+%prod.quarkus.datasource.jdbc.url=jdbc:postgresql://${{ values.database_name }}-postgresql:5432/${{ values.database_name }}
+%prod.quarkus.datasource.username=${{ values.database_username }}
 %prod.quarkus.datasource.password=${{ values.database_password }}
```
Comment (iocanel) [inline qshift/templates/quarkus-application/skeletons/quarkus-orm-rest/src/main/resources/application.properties:10]: We should not pass the password to the application.properties

File: qshift/templates/quarkus-application/manifests/argocd/${{values.component_id}}-db.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    namespace: ${{ values.appNamespace }}
+  source:
+    chart: postgresql
+    repoURL: https://charts.bitnami.com/bitnami
+    targetRevision: 11.9.13
+    helm:
+      values: |
+        auth:
+          username: ${{ values.database_username }}
+          password: ${{ values.database_password }}
```
Comment (iocanel) [inline qshift/templates/quarkus-application/manifests/argocd/${{values.component_id}}-db.yaml:22]: we should probably not pass the password here and let helm generate one.

**Review by iocanel — CHANGES_REQUESTED** [review]

File: qshift/templates/quarkus-application/manifests/argocd/${{values.component_id}}-db.yaml (hunk 3 above)
Comment (cmoulliard) [reply qshift/templates/quarkus-application/manifests/argocd/${{values.component_id}}-db.yaml:22]: This is correct. I missed to remove it here ;-). Will fix it

File: qshift/templates/quarkus-application/skeletons/quarkus-orm-rest/src/main/resources/application.properties (hunk 2 above)
Comment (cmoulliard) [reply qshift/templates/quarkus-application/skeletons/quarkus-orm-rest/src/main/resources/application.properties]: Fixed: d84d675

**Review by iocanel — APPROVED** [review]
LGTM

**Pull request merged**